6-more-secure-builds/results.md

```diff
@@ -69,4 +73,43 @@ a348eb9fe0c3   About a minute ago   CMD ["/bin/sh" "-c" "node src/index.js"]
 <missing>      2 weeks ago          ENV NODE_VERSION=20.17.0                        0B        buildkit.dockerfile.v0
 <missing>      6 weeks ago          /bin/sh -c #(nop)  CMD ["/bin/sh"]              0B        
 <missing>      6 weeks ago          /bin/sh -c #(nop) ADD file:a71f7e9bc66668361…   8.83MB 
+```
+Interestingly, using command 'COPY . .' in Dockerfile.multistage both before and after 'npm ci' allows the image to be built without
+any errors and for it to run correctly, suggesting something isn't quite right.
+
+Trying to copy just the node_modules has been failing, could this be because they are included in the .dockerignore?
+
+After commenting out node_modules from .dockerignore file we're then able to copy those over, meaning we can take only the dependencies we need and 
```

Comment:
So here's a gotcha, in many languages, dependencies can have a link to the underlying architecture. If you have done the npm ci on a mac, should you use those modules or should you load the modules fresh on the docker image architecture (which is likely to be linux).
What files do you need on the first stage to enable the npm ci to work fresh on the first stage?

Reply:
I feel like I need the package.json file, and after some reading, it seems recommended that the package-lock.json file is also included. We also need our .npmrc file as this is where our PAT is being read to give us permission to install these dependencies.  

I could run an experiment and try to build this image on my personal machine, which is running linux and see if it blows up. This could be interesting...